src/main/java/org/folio/rest/impl/StagingUsersAPI.java

```diff
@@ -0,0 +1,109 @@
+
+  private static void updateMetaInfo(Map<String, String> okapiHeaders, StagingUser existingStagingUser) {
+    // updating metadata
+    Metadata metadata = existingStagingUser.getMetadata();
+    metadata.setUpdatedDate(new Date());
+    metadata.setUpdatedByUserId(MetadataUtil.createMetadata(okapiHeaders).getUpdatedByUserId());
+    existingStagingUser.setMetadata(metadata);
+  }
+
+  @Override
+  public void getStagingUsers(String query, String orderBy, StagingUsersGetOrder order, String totalRecords, int offset,
+                              int limit, Map<String, String> okapiHeaders,
+                              Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    logger.debug("getStagingUsers:: query: {}, orderBy: {}, order: {}, totalRecords: {}, offset: {}, limit: {}",
+      query, orderBy, order, totalRecords, offset, limit);
+    PgUtil.get(STAGING_USERS_TABLE, StagingUser.class, StagingUserdataCollection.class,
+      query, offset, limit, okapiHeaders, vertxContext, StagingUsers.GetStagingUsersResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  public void postStagingUsers(StagingUser entity, Map<String, String> okapiHeaders,
+                               Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    logger.debug("postStagingUsers:: request body: {}", entity);
+    try {
+      PostgresClient postgresClient = PgUtil.postgresClient(vertxContext, okapiHeaders);
+      final Criterion criterion = new Criterion(
+        new Criteria().addField("'contactInfo'").addField("'email'")
+          .setOperation("=").setVal(entity.getContactInfo().getEmail()).setJSONB(true));
+      AtomicReference<Boolean> isUpdated = new AtomicReference<>(Boolean.FALSE);
+      postgresClient.withTrans(conn -> conn.get(STAGING_USERS_TABLE, StagingUser.class, criterion, true)
+        .compose(stagingUserResults -> {
+          logger.info("Record found by email: {} ", stagingUserResults.getResults().size());
+          List<StagingUser> stagingUsersByEmail = stagingUserResults.getResults();
+
+          String entityId = null;
+          if (stagingUsersByEmail != null && !stagingUsersByEmail.isEmpty()) {
+            StagingUser existingStagingUser = stagingUsersByEmail.get(0);
+            entityId = existingStagingUser.getId();
+            logger.info("Processing existing staging user with ID: {}", existingStagingUser.getId());
+
+            // Copy non-null properties
+            BeanUtilsExtended.copyPropertiesNotNull(existingStagingUser, entity);
+            entity.setIsEmailVerified(existingStagingUser.getIsEmailVerified());
+
+            updateMetaInfo(okapiHeaders, existingStagingUser);
+            isUpdated.set(Boolean.TRUE);
+          } else {
+            logger.info("Creating new Staging-User");
+            entity.setStatus(StagingUser.Status.TIER_1);
+            entity.setIsEmailVerified(Boolean.FALSE);
```

Comment:
IF the payload does not contain isEmailVerified value then set to false. Otherwise use the payload value.

Reply:
isEmailVerifed !=null ? entity.getEmailVerfiied() : False